service/test/ConstConfigIOGroupTest.cpp

```diff
@@ -0,0 +1,565 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "ConstConfigIOGroup.hpp"
+
+#include <set>
+#include <string>
+#include <fstream>
+#include <iostream>
+#include <regex>
+#include <unistd.h>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "geopm_topo.h"  // for geopm_domain_e
+#include "geopm_test.hpp" // for GEOPM_EXPECT_THROW_MESSAGE
+
+#include "geopm/Helper.hpp"
+#include "geopm/Agg.hpp"
+#include "geopm/Exception.hpp"
+#include "geopm/PlatformIO.hpp"
+#include "geopm/PlatformTopo.hpp"
+
+using geopm::ConstConfigIOGroup;
+using geopm::Exception;
+using geopm::PlatformIO;
+using geopm::PlatformTopo;
+
+class ConstConfigIOGroupTest : public::testing::Test
+{
+    protected:
+        void SetUp() override;
+        void TearDown() override;
+};
+
+void ConstConfigIOGroupTest::SetUp()
+{
+}
+
+TEST_F(ConstConfigIOGroupTest, input_empty_string)
+{
+    GEOPM_EXPECT_THROW_MESSAGE(
+        ConstConfigIOGroup iogroup(""),
+        GEOPM_ERROR_INVALID,
+        "ConstConfigIOGroup::parse_config_json(): "
+        "detected a malformed JSON string"
+    );
+}
+
+TEST_F(ConstConfigIOGroupTest, input_empty_json)
+{
+    ConstConfigIOGroup iogroup("{}");
+    EXPECT_EQ(iogroup.signal_names(), std::set<std::string>{});
+}
+
+TEST_F(ConstConfigIOGroupTest, input_gibberish)
+{
+    GEOPM_EXPECT_THROW_MESSAGE(
+        ConstConfigIOGroup iogroup("asdfklfj234890fnjklsd"),
+        GEOPM_ERROR_INVALID,
+        "ConstConfigIOGroup::parse_config_json(): "
+        "detected a malformed JSON string"
+    );
+}
+
+TEST_F(ConstConfigIOGroupTest, input_duplicate_signal)
+{
+    /* When one signal is provided more than once,
+       apparently the latter entry overwrites the data of the former. */
```

Comment:
Suggested change in comment: When one signal is provided more than once, the latest entry will take precedence. This is due to how the underlying JSON library handles duplicate entries.